packages/codegen-ui-react/lib/forms/react-form-renderer.ts

```diff
@@ -434,23 +437,44 @@ export abstract class ReactFormTemplateRenderer extends StudioTemplateRenderer<
       }
     }
 
-    statements.push(resetStateFunction(formMetadata.fieldConfigs, defaultValueVariableName));
+    const hasManyFieldConfigs = getHasManyFieldConfigs(formMetadata.fieldConfigs);
+    statements.push(resetStateFunction(formMetadata.fieldConfigs, defaultValueVariableName, hasManyFieldConfigs));
 
+    const linkedDataNames: string[] = [];
     if (isDataStoreUpdateForm) {
       statements.push(
         buildUseStateExpression(lowerCaseDataTypeNameRecord, factory.createIdentifier(lowerCaseDataTypeName)),
       );
-      statements.push(
-        addUseEffectWrapper(
-          buildUpdateDatastoreQuery(modelName, lowerCaseDataTypeNameRecord),
-          // TODO: change once cpk is supported in datastore
-          ['id', lowerCaseDataTypeName],
-        ),
-      );
+      if (hasManyFieldConfigs.length > 0) {
+        hasManyFieldConfigs.forEach((hasManyFieldConfig) => {
+          const [fieldName] = hasManyFieldConfig;
+          const linkedDataName = getLinkedDataName(fieldName);
+          linkedDataNames.push(linkedDataName);
+          statements.push(buildUseStateExpression(linkedDataName, factory.createIdentifier('[]')));
+        });
+      }
+
+      if (hasManyFieldConfigs.length > 0) {
```

Comment:
Is this intentionally? looks like the condition is the same as above. Can we combine these two conditionals?

Reply:
Oh right, i can refine this part a bit by combing those